### PR TITLE
Add input_spec & output_spec for TranslatedLayer

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

New features
### PR changes

APIs
### Describe

Add `TranslatedLayer._input_spec & _output_spec` for paddle2onnx

General users do not need to obtain the description of the input and output, so these two method temporarily used as internal apis.

### CODE REVIEW

**Review by liym27 — DISMISSED** [review]
LGTM

**Comment by Channingss** [comment]
LGTM

File: python/paddle/fluid/dygraph/io.py [hunk 1]
```diff
@@ -913,15 +915,52 @@ def train(layer, loader, loss_fn, opt):
...
 
         # 3. construct program
         program = _build_program_by_desc(program_desc)
         return program
+
+    def _get_program_holder(self, method_name='forward'):
+        program_holder = self._program_holder_dict.get(method_name, None)
+        if program_holder is None:
+            raise ValueError(
+                "The method `%s` is not exists in loaded TranslatedLayer." %
```
Comment (zhhsplendid) [inline python/paddle/fluid/dygraph/io.py]: Grammar issue:

"is not exists" -> "does not exist"

File: python/paddle/fluid/dygraph/io.py [hunk 2]
```diff
@@ -913,15 +915,52 @@ def train(layer, loader, loss_fn, opt):
...
+                method_name)
+        return program_holder
+
+    def _input_spec(self, method_name='forward'):
```
Comment (Channingss) [inline python/paddle/fluid/dygraph/io.py:935]: w

**Review by zhhsplendid — COMMENTED** [review]
Overall LGTM, please note my only comment

**Review by Channingss — DISMISSED** [review]
LGTM

**Review by Aurelius84 — DISMISSED** [review]

File: python/paddle/fluid/dygraph/io.py (hunk 1 above)
Comment (chenwhql) [reply python/paddle/fluid/dygraph/io.py]: done, thx

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Pull request merged**